apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/session.ex

```diff
@@ -0,0 +1,232 @@
+defmodule RigInboundGatewayWeb.Session do
+  @moduledoc """
+  Maintains subscriptions and buffers incoming events.
+
+  Started by, or connected from, a (longpolling) connection process.
+  """
+  use GenServer
+
+  require Logger
+
+  alias Rig.EventFilter
+  alias RIG.Subscriptions
+  alias RigCloudEvents.CloudEvent
+  alias RigInboundGateway.Events
+  alias RigInboundGatewayWeb.EventBuffer
+
+  @window_size_ms 500
+  @max_tries 10
+  @event_buffer_size 200
+  @subscription_refresh_interval_ms 60_000
+  @session_timeout_validation_interval_ms 60_000
+  @session_timeout_ms 3_600_000
+
+  # ---
+  def start(query_params, opts \\ []) do
+    GenServer.start(__MODULE__, %{query_params: query_params}, opts)
+  end
+
+  def recv_events(server, last_event_id) do
+    GenServer.call(server, {:recv_events, last_event_id || "first_event", 0}, 20_000)
```

Comment:
if you accept my changes to EventBuffer, this hardcoded "first_event" should be removed